mlflow/genai/optimize/types.py

```diff
@@ -12,22 +12,62 @@
 @experimental
 @dataclass
 class PromptOptimizationResult:
+    """
+    Result of prompt optimization.
+
+    This class represents the result of :py:func:`mlflow.genai.optimize_prompt()`.
+    The optimized prompt template is automatically registered as a new version of the
+    original prompt and included in the result.
```

Comment:
Do we need to explain what `optimize_prompt` does here?

Reply:
I thought so since otherwise what `PromptOptimizationResult.prompt` means is not clear.